src/plugins/memdump/memdump.cpp

```diff
@@ -360,6 +360,59 @@ bool dump_memory_region(
     return ret;
 }
 
+static bool dump_if_points_to_executable_memory(
+    drakvuf_t drakvuf,
+    drakvuf_trap_info* info,
+    vmi_instance_t vmi,
+    addr_t process_base,
+    addr_t target_addr,
+    const char* reason,
+    extras_t *extras)
+{
+    memdump* plugin = get_trap_plugin<memdump>(info);
+    if (!plugin)
+        return false;
+
+    addr_t dtb;
+    if (!drakvuf_get_process_dtb(drakvuf, process_base, &dtb)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve process dtb\n");
+        return false;
+    }
+
+    // Get page protection flags.
+    page_info_t p_info = {};
+    if (VMI_SUCCESS != vmi_pagetable_lookup_extended(vmi, dtb, target_addr, &p_info)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to retrieve page protection flags\n");
+        return false;
+    }
+
+    bool page_valid = (p_info.x86_ia32e.pte_value & (1UL << 0)) != 0;
+    bool page_execute = (p_info.x86_ia32e.pte_value & (1UL << 63)) == 0;
+    if (!page_valid || !page_execute) {
+        return false;
+    }
+
+    // Segment is valid and executable – dump it.
+    mmvad_info_t mmvad;
+    if (!drakvuf_find_mmvad(drakvuf, process_base, target_addr, &mmvad)) {
+        PRINT_DEBUG("[MEMDUMP] Failed to find mmvad\n");
+        return false;
+    }
+
+    access_context_t ctx {
+        .translate_mechanism = VMI_TM_PROCESS_DTB,
+        .dtb = dtb,
+        .addr = mmvad.starting_vpn * VMI_PS_4KB
+    };
+    size_t dump_size = (mmvad.ending_vpn - mmvad.starting_vpn + 1) * VMI_PS_4KB;
+    bool print_extras = extras ? true : false;
+    if (!dump_memory_region(drakvuf, vmi, info, plugin, &ctx, dump_size, reason, extras, print_extras)) {
```

Comment:
```suggestion
    if (!dump_memory_region(drakvuf, vmi, info, plugin, &ctx, dump_size, reason, extras, extras == nullptr)) {
```
We could use some implicit conversion magic here, but this is more verbose :)